go/vt/vtctl/reparentutil/util.go

```diff
@@ -52,9 +52,6 @@ func ChooseNewPrimary(
 	// (TODO:@ajm188) it's a little gross we need to pass this, maybe embed in the context?
 	logger logutil.Logger,
 ) (*topodatapb.TabletAlias, error) {
-	if avoidPrimaryAlias == nil {
-		return nil, vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "tablet to avoid for reparent is not provided, cannot choose new primary")
-	}
```

Comment:
What case was this check trying to handle?

Reply:
Earlier, even when the user did not provide an `avoidPrimaryAlias` we used to set it to the shard's current primary and so never called `ChooseNewPrimary` with a nil `avoidPrimaryAlias`. But this had to be changed for the case of initialisation wherein we can choose  a new primary and there is not tablet which we should be avoiding. So, this check for a nil avoidPrimaryAlias has been removed and test cases for calling `ChooseNewPrimary` with `avoidPrimaryAlias` nil have been added.

Reply:
This change was indeed not correct and is the cause of the issue https://github.com/vitessio/vitess/issues/12018. However the problem that this fix intended to solve was still a concern. All of this has been addressed in https://github.com/vitessio/vitess/pull/12019